### PR TITLE
libpod: Podman info output more network information

### PR DESCRIPTION
**Pull request by sstosh**
podman info prints the network information about binary path,
package version, program version and DNS information.

Fixes: #18443

<details>
<summary>cni info</summary>

```yaml
  networkBackendInfo:
    backend: cni
    dns:
      package: podman-plugins-4.5.0-1.fc38.x86_64
      path: /usr/libexec/cni/dnsname
      version: |-
        CNI dnsname plugin
        version: 1.3.1
        commit: unknown
    package: |-
      containernetworking-plugins-1.1.1-16.fc38.x86_64
      podman-plugins-4.5.0-1.fc38.x86_64
    path: /usr/libexec/cni
```
</details>

<details>
<summary>netavark info</summary>

```yaml
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.6.0-1.fc38.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.6.0
    package: netavark-1.6.0-2.fc38.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.6.0
```
</details>

#### Does this PR introduce a user-facing change?

```release-note
Now podman info prints the network information about binary path, package version, program version and DNS information.
```

### CODE REVIEW

File: libpod/define/info.go [hunk 1]
```diff
@@ -26,26 +26,26 @@ type SecurityInfo struct {
...
+	DatabaseBackend   string                 `json:"databaseBackend"`
+	Distribution      DistributionInfo       `json:"distribution"`
+	EventLogger       string                 `json:"eventLogger"`
+	Hostname          string                 `json:"hostname"`
+	IDMappings        IDMappings             `json:"idMappings,omitempty"`
+	Kernel            string                 `json:"kernel"`
+	LogDriver         string                 `json:"logDriver"`
+	MemFree           int64                  `json:"memFree"`
+	MemTotal          int64                  `json:"memTotal"`
+	NetworkBackend    map[string]interface{} `json:"networkBackend"`
```
Comment (Luap99) [inline libpod/define/info.go:46]: This is a breaking change and thus cannot be made, I suggest adding a new NetworkBackendInfo field.

**Review by Luap99 — CHANGES_REQUESTED** [review]
Thanks for doing this, I really like the idea. However I think it would be much better to move the logic directly into the networkBackend in c/common and then add a function there called Info() to return the info you want. Alo I would rather have a proper struct with the info then a generic map.

You can move `packageVersion` and `programVersion` to some c/common util package if needed.

**Comment by baude** [comment]
@sstosh what say you?

**Comment by sstosh** [comment]
> Just adding a note but totally upto @sstosh , once we have proper implementation in `c/common` as discussed above, I think PR could be easily extended to also include `aardvark-dns` details. Then this will close: #18443

@flouthoc  OK. I also will try to add the feature to show the details of aardvark-dns.

**Comment by sstosh** [comment]
The output is as shown below:

```yaml
  networkBackend: netavark
  networkBackendInfo:
    aardvark-package: aardvark-dns-1.6.0-1.fc37.x86_64
    aardvark-path: /usr/libexec/podman/aardvark-dns
    aardvark-version: aardvark-dns 1.6.0
    netavark-package: netavark-1.6.0-2.fc37.x86_64
    netavark-path: /usr/libexec/podman/netavark
    netavark-version: netavark 1.6.0
```

```yaml
  networkBackend: cni
  networkBackendInfo: null
```

**Comment by flouthoc** [comment]
@sstosh Thank you :smile: output LGTM

**Comment by sstosh** [comment]
The review is ready.

**Review by flouthoc — APPROVED** [review]
LGTM Thanks @sstosh

**Review by Luap99 — APPROVED** [review]
LGTM, thank you

**Pull request merged**